### DISCUSSION TITLE
JSON in source

### DISCUSSION THREAD

**Discussion by Datadroves** [vercel/next.js#17640] [Help] [answered]
Hi,

Is there a way to prevent JSON from being rendered on the page? I'm able to see the entire response from our API in the source, including the MongoDB IDs. Is there a workaround for this?

We're using getServerSideProps with fetch to call the API route.

Here's the URL - https://bit.ly/2GocH73

**Comment by paulogdm** [comment]
Hello :wave: 

I've transferred this issue to the Next.js repository since it seems to be a framework-specific question. Can you give us more context (code) around the issue and also some code context?

**Comment by jamesmosier** [comment, marked as answer]
If you return the data as `props` from `getServerSideProps` then it will be rendered to the DOM as JSON. This is how React/Next knows what data to render during server rendering. It has to pull that information from somewhere to construct the DOM.

If there is any sensitive data you don't want exposed to the client, you should be pruning that from being returned in `props`.
